BUILD.openssl.bazel

```diff
@@ -26,41 +26,46 @@ MAKE_TARGETS = [
 ]
 
 config_setting(
-    name = "msvc_compiler",
+    name = "clang_cl_compiler",
     flag_values = {
-        "@bazel_tools//tools/cpp:compiler": "msvc-cl",
+        "@bazel_tools//tools/cpp:compiler": "clang-cl",
     },
     visibility = ["//visibility:public"],
 )
 
 alias(
     name = "openssl",
     actual = select({
-        ":msvc_compiler": "openssl_msvc",
+        ":clang_cl_compiler": "openssl_clang_cl",
         "//conditions:default": "openssl_default",
     }),
     visibility = ["//visibility:public"],
 )
 
 configure_make_variant(
-    name = "openssl_msvc",
+    name = "openssl_clang_cl",
     build_data = [
         "@nasm//:nasm",
         "@perl//:perl",
     ],
     configure_command = "Configure",
     configure_in_place = True,
     configure_options = CONFIGURE_OPTIONS + [
+        # Target
         "VC-WIN64A",
-        # Unset Microsoft Assembler (MASM) flags set by built-in MSVC
-        # toolchain, as NASM is used to build OpenSSL rather than MASM.
-        "ASFLAGS=\" \"",
+        # Clean env vars set by previous bazel targets.
+        # Perl script needs generated PDB file to run successfully.
+        # clang-cl does support generating PDB, but we couldn't
+        # figure out why it didn't make one.
```

Comment:
Let's be very professional: instead of leaving it at "we couldn't figure out why [something]", let's consider this "unfinished business that's not currently a priority". The tech didn't beat us! We just don't think it's important to beat the tech right now! 😇 

To do so, we do two things:
1. Let's file a GitHub issue on this repo (which we won't schedule time to deal with for the moment) with a title like "Investigate whether OpenSSL can build with clang-cl". Give it a short description of the current state of the investigation (also see below).
2. Reword this comment to express the same sentiment, with something like...
```suggestion
        # Override environment variables set by higher-level Bazel targets.
        # Most importantly, set the C++ compiler to be `cl` (as opposed to 
        #`clang-cl`). We're unsure whether OpenSSL can be built using
        # clang-cl. 
        # TODO(#NUMBER_OF_ISSUE): Investigate whether OpenSSL can
        # build with clang-cl.
```

You can add information as to _why_ we can't get it to build with `clang-cl` in the issue, with a line like...
> OpenSSL's build script expects a .PDB file to be generated. While clang-cl does support generating PDB files, we haven't yet found the right combination of flags to make it do so.

That has the nice benefit that as the investigation continues in the future, we can keep updating the issue to have the latest information, without having to update code comments.